fission-core/library/Fission/User/Username/Types.hs

```diff
@@ -47,8 +51,18 @@ instance ToHttpApiData Username where
 instance PersistField Username where
   toPersistValue (Username un) = PersistText un
   fromPersistValue = \case
-    PersistText un -> Right (Username un)
-    other          -> Left ("Invalid Persistent Username: " <> Text.pack (show other))
+    input@(PersistText txt) ->
+      -- NOTE this may break against old DB records, but we really should keep this strict
+      -- I'd rather fail fast so know about it so we can fix it ~expede
```

Comment:
This is the controversial bit. Basically, don't trust what's in the DB, even if that breaks old bad records. This is guarding against things like us trying to use non-URL-safe characters for DNS.